src/app/saga.ts

```diff
@@ -415,6 +416,11 @@ export function* handleSetAppState(action: SetAppState) {
   if (requirePinOnAppOpen && isPassedDoNotLockPeriod && isAppActive) {
     yield put(appLock())
   }
+
+  if (isAppActive) {
+    // Force a statsig refresh by updating the user object
+    yield call(patchUpdateStatsigUser)
```

Comment:
it seems like patchUpdateStatsigUser was created to work around an assumption we had about how Statsig updateUser was working that we later found to be false-- namely, that updateUser wipes the current user properties (we found this only to be the case when you update userId). I am wondering if we should refactor patchUpdateStatsigUser to make it more single-purpose to updating the userId, and call another function here that is also single-purpose. 

Relatedly, I find it pretty surprising that you can just call patchUpdateStatsigUser here and be done with it. Both that meta-observation and the quite sizable docstring on the function indicate to me that its behavior is rather opaque and non-intuitive at the moment, which further motivate a refactor in my view.

Reply:
> we found this only to be the case when you update userId
I don't think this is true unless I am missing something. AFAIK, updateUser requires the full object in all cases. The only difference is that if a userId is updated, it does a full reinitialization, whereas when other properties are updated, statsig returns cached values while this is still resolving, and if we call updateUser with no userId, it assumes a null user id.
But I agree on the method docstring, we could maybe rename it to maybe refreshStatsig and take optional user properties?